yarn-project/end-to-end/src/e2e_key_rotation.test.ts

```diff
@@ -0,0 +1,233 @@
+import { createAccounts } from '@aztec/accounts/testing';
+import {
+  type AztecAddress,
+  type AztecNode,
+  type DebugLogger,
+  ExtendedNote,
+  Fq,
+  Fr,
+  Note,
+  type PXE,
+  type TxHash,
+  type Wallet,
+  computeSecretHash,
+  retryUntil,
+} from '@aztec/aztec.js';
+import { type PublicKey, derivePublicKeyFromSecretKey } from '@aztec/circuits.js';
+import { KeyRegistryContract, TestContract, TokenContract } from '@aztec/noir-contracts.js';
+import { getCanonicalKeyRegistryAddress } from '@aztec/protocol-contracts/key-registry';
+
+import { jest } from '@jest/globals';
+
+import { expectsNumOfEncryptedLogsInTheLastBlockToBe, setup, setupPXEService } from './fixtures/utils.js';
+
+const TIMEOUT = 120_000;
+
+const SHARED_MUTABLE_DELAY = 5;
+
+describe('e2e_key_rotation', () => {
+  jest.setTimeout(TIMEOUT);
+
+  let aztecNode: AztecNode;
+  let pxeA: PXE;
+  let pxeB: PXE;
+  let walletA: Wallet;
+  let walletB: Wallet;
+  let logger: DebugLogger;
+  let teardownA: () => Promise<void>;
+  let teardownB: () => Promise<void>;
+
+  let keyRegistryWithB: KeyRegistryContract;
+  let testContract: TestContract;
+  let contractWithWalletA: TokenContract;
+  let contractWithWalletB: TokenContract;
+
+  let tokenAddress: AztecAddress;
+
+  const initialBalance = 987n;
+
+  beforeAll(async () => {
+    ({
+      aztecNode,
+      pxe: pxeA,
+      wallets: [walletA],
+      logger,
+      teardown: teardownA,
+    } = await setup(1));
+
+    ({ pxe: pxeB, teardown: teardownB } = await setupPXEService(aztecNode, {}, undefined, true));
+
+    [walletB] = await createAccounts(pxeB, 1);
+    keyRegistryWithB = await KeyRegistryContract.at(getCanonicalKeyRegistryAddress(), walletB);
+
+    // We deploy test and token contracts
+    testContract = await TestContract.deploy(walletA).send().deployed();
+    const tokenInstance = await deployTokenContract(initialBalance, walletA.getAddress(), pxeA);
+    tokenAddress = tokenInstance.address;
+
+    // Add account B to wallet A
+    await pxeA.registerRecipient(walletB.getCompleteAddress());
```

Comment:
Can we move this up so we are clearly marking it. The errors I'm getting in #6432 seems to be because of a not registered account or something which seems like a fairly common potential issue. I expect that we have the same issue in other places already, but for some yet unknown reason, fetching 0,0 seems to not fail with the ts 🤷